### DISCUSSION TITLE
Avoiding a PR entirely when artifact updates fail

### DISCUSSION THREAD

**Discussion by JennToo** [renovatebot/renovate#18268] [Request Help] [answered]
# Background of the problem

Some context for this issue:

- We self-host Renovate, and just run it against repos on a timer
- We pull all of our npm dependencies through an internal Artifactory mirror

We've recently started seeing an issue like this on some of our npm projects:

![image](https://user-images.githubusercontent.com/65889/195126531-f9e89557-703b-4146-9a78-f5215be20681.png)
------
![image](https://user-images.githubusercontent.com/65889/195126567-ca153525-3034-4234-bd79-3b3059f1aad4.png)

My suspicion is that our internal Artifactory instance isn't always providing a consistent view of the npm upstream. So when a batch of packages that have inter-dependencies releases to npm, there's a period of time where artifact updates will fail when looking at our own instance.

# How I think we could work around it

In an ideal world, our Artifactory instance would just provide consistent information about the npm registry. But I don't think that's going to happen.

I was looking through the Renovate docs though trying to determine if there's any way to just get Renovate to not make a PR at all if the artifact update fails. That way we wouldn't need to go manually rebase them when this happens. Eventually our internal infrastructure will scan the repo with Renovate again, and eventually the update will start working.

There's the `suppressNotifications` option in the docs, which seems sorta close to what we're looking for.

Anyone have thoughts on how this could be achieved? Or maybe have other ideas about what's causing the problem in the first place?

**Comment by rarkins** [comment, marked as answer]
There's already one measure in place for similar reasons:
- If artifacts updating fails, and a release is less than 24 hours old, then Renovate doesn't create the branch/PR

I'm not certain it's still working, and also it may not cover all use cases - for example you have an existing branch/PR already.

Avoiding the PR entirely is not the best idea unless you are watching your dependency dashboard too. Otherwise you could end up with permanently suppressed updates that you're unaware of

**Reply by JennToo** [reply]
Ah, thanks for the tip!

Definitely agreed about putting off the PR for forever, that's definitely not ideal.

But the holding period sounds like it could work. I couldn't find the 24 hour artifact-update-failure hold feature in the source code. This was the closest thing I could find:
https://github.com/renovatebot/renovate/blob/b9cfb98219489619ebf660975c5a0c0d6b62b795/lib/workers/repository/update/branch/index.ts#L441-L445

However, digging through the function above I did spot the [`stabilityDays` option][1], which might could give us similar behavior. We don't need mega fast turnaround for updating most npm packages, so I think just setting this option to `1` would be a good enough solution for us.

And thanks again as always for your work on Renovate! It's been a fantastic tool for us

[1]: https://docs.renovatebot.com/configuration-options/#stabilitydays
